### PR TITLE
refactor(components): [calendar] refactor

### PR DESCRIPTION
**Pull request by buqiyuan**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/calendar/src/date-table.ts [hunk 1]
```diff
@@ -1,6 +1,40 @@
 import { buildProps, definePropType, isObject } from '@element-plus/utils'
+import { rangeArr } from '@element-plus/components/time-picker'
 import type { ExtractPropTypes } from 'vue'
 import type { Dayjs } from 'dayjs'
+import type DateTable from './date-table.vue'
+
+export type CellType = 'next' | 'prev' | 'current'
+export type Cell = {
```
Comment (sxzz) [inline packages/components/calendar/src/date-table.ts:8]: Same here

File: packages/components/calendar/src/date-table.ts [hunk 2]
```diff
@@ -1,6 +1,40 @@
...
+  text: number
+  type: CellType
+}
+
+export const WEEK_DAYS = [
```
Comment (sxzz) [inline packages/components/calendar/src/date-table.ts:13]: We can move it to `@element-plus/constants/date-pick.ts` (and maybe rename `date-pick.ts` to `date.ts`)

File: packages/components/calendar/__tests__/calendar.spec.tsx [hunk 3]
```diff
@@ -2,28 +2,16 @@ import { nextTick } from 'vue'
 import { mount } from '@vue/test-utils'
```
Comment (sxzz) [inline packages/components/calendar/__tests__/calendar.spec.tsx:2]: Please remove `as any` as possible as you can.
https://github.com/element-plus/element-plus/pull/6682/files#diff-4cff895ccf7da8859ac2d4e5300d00dc9ea5bf00cbc3ef77427e14e8fce60ff1R24

**Comment by buqiyuan** [comment]
@sxzz all done

**Review by jw-foss — APPROVED** [review]

**Pull request merged**